scripts/cromwell/analyze_resource_acquisition.py

```diff
@@ -0,0 +1,460 @@
+#!/bin/python
+
+import json
+import argparse
+import numpy as np
+import dateutil.parser
+from dateutil.tz import tzutc
+import datetime
+import time
+import pandas as pd
+import math
+import random
+import matplotlib.pyplot as plt
+import matplotlib.ticker
+import sys
+from os.path import basename, isfile, getsize
+SEED = 42
+
+"""
+Summary: Scrapes workflow metadata to analyze resource acquisition (VMs, CPU, RAM, disk memory), for the purpose
+	of understanding resource peaks and timing jobs to avoid hitting quotas.
+
+Usage: 
+	python analyze_resource_acquisition workflow_metadata.json /path/to/output_basename [optional flags]
+	Optional flags:
+	--plot_title WorkflowName (WorkflowName will be prepended to plot title)
+	--save_table (to save a TSV file of the table used to create the plot)
+	--override_warning (to attempt to run script on workflow that raised a warning)
+
+Parameters:
+	workflow_metadata.json: path to Cromwell metadata file for workflow of interest
+	/path/to/output_basename: base output path, to which extensions will be appended for each output file, 
+		ie. /output_dir/basename will yield /output_dir/basename.plot.png, /output_dir/basename.table.tsv, etc
+		or, /output_dir/ will yield plot.png, table.tsv, peaks.txt, etc
+
+Outputs:
+	- plot (png) of VMs, CPUs (total, preemptible, and nonpreemptible), RAM, and disk (HDD, SSD) acquisitioned over time 
+	- (optional, with --save_table flag) table of resource acquisition over time for each type of resource listed above
+	- peak resource acquisition for each type of resource listed above (TSV file)
+	- number of non-preemptible VMs used (prints to stdout) and the names of the tasks that used them (TSV file, if any)
+	- number of tasks call-cached (prints to stdout), and their task names (TSV file, if any)
+"""
+NUM_CACHED = 0
+CACHED = dict()
+NUM_NONPREEMPTIBLE = 0
+NONPREEMBTIBLE_TASKS = dict()
```

Comment:
For the future, it is better to avoid global variables whenever possible. It would be better to pass these around as local variables. This makes your code easier to read, more portable, and amenable to unit testing. An alternative might be to wrap the script into a class and have these be member variables.

Reply:
I agree, but it would not have been straightforward to pass these variables as local variables given the recursive nature of the `getCalls()` function. I can revisit this, maybe try a class, if needed